### PR TITLE
add CODECOV workflow with code coverage stats

### PR DESCRIPTION
**Pull request by bieli**

### CODE REVIEW

File: .github/workflows/codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Codecov workflow
+on: [push]
+jobs:
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
```
Comment (przemek1236) [inline .github/workflows/codecov.yaml:9]: This makes the workflow to run for each python version. Wondering if this is needed at this point 🤔

Comment (bieli) [reply .github/workflows/codecov.yaml:9]: It's good to know, if something specific like bug in code will be connected with particular Python version and OS. This CI gives us red flag, so I think we no need additional multiplatform verifications for unit tests.

File: .github/workflows/codecov.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Setup Python
+      uses: actions/setup-python@master
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Generate coverage report
+      run: |
+        pip install pytest
+        pip install pytest-cov
+        pip install -r requirements-dev.txt
+        pytest --cov=./ --cov-report=xml
```
[the remark this answers is not in the document]
Comment (bieli) [reply .github/workflows/codecov.yaml:24]: Yeah, good idea!

[the remark this answers is not in the document]
Comment (bieli) [reply .github/workflows/codecov.yaml:24]: @przemek1236 Would you like to change this?

[the remark this answers is not in the document]
Comment (przemek1236) [reply .github/workflows/codecov.yaml:24]: Sure, I can to over this change

**Pull request merged**